composable_kernel/include/tensor_operation/gridwise_2d_reduction_blockwise.hpp

```diff
@@ -0,0 +1,863 @@
+    };
+};
+
+template <typename GridwiseReduction,
+          bool NeedIndices,
+          typename InDataType,
+          typename OutDataType,
+          typename AccDataType,
+          typename In2dDescType,
+          typename Out1dDescType,
+          typename InElementwiseOperation,
+          typename OutElementwiseOperation>
+__global__ void
+kernel_reduce_blockwise_second_call(const In2dDescType in2dDesc,
+                                    const Out1dDescType out1dDesc,
+                                    const InElementwiseOperation in_elementwise_op,
+                                    const OutElementwiseOperation acc_elementwise_op,
+                                    AccDataType alpha,
+                                    const InDataType* const __restrict__ p_src_global,
+                                    OutDataType beta,
+                                    OutDataType* const __restrict__ p_dst_global,
+                                    const int* const __restrict__ ws_indices_global,
+                                    int* const __restrict__ indices_global)
+{
+    if constexpr(!NeedIndices)
+    {
+        GridwiseReduction::Run(in2dDesc,
+                               out1dDesc,
+                               in_elementwise_op,
+                               acc_elementwise_op,
+                               alpha,
+                               p_src_global,
+                               beta,
+                               p_dst_global,
+                               ws_indices_global,
+                               indices_global);
+    }
+    else
+    {
+        GridwiseReduction::RunSecondCallWithIndices(in2dDesc,
+                                                    out1dDesc,
+                                                    in_elementwise_op,
+                                                    acc_elementwise_op,
+                                                    alpha,
+                                                    p_src_global,
+                                                    beta,
+                                                    p_dst_global,
+                                                    ws_indices_global,
+                                                    indices_global);
+    };
+};
+
+template <typename InDataType,
+          typename OutDataType,
+          typename AccDataType,
+          typename In2dDescType,
+          typename Out1dDescType,
+          typename ReduceOperation,
+          typename InElementwiseOperation,
+          typename OutElementwiseOperation,
+          bool PropagateNan,
+          bool BetaIsZero,
+          index_t BlockSize,
+          index_t MThreadClusterSize,
+          index_t KThreadClusterSize,
+          index_t MThreadSliceSize,
+          index_t KThreadSliceSize,
+          index_t InSrcVectorDim,
+          index_t InSrcVectorSize,
+          index_t OutDstVectorSize>
+struct GridwiseReduction_xy_to_x_blockwise
```

Comment:
please choose a name for each of the dimension, and then be consistent in naming the dimension in all the code
 for example:
M: for non-reduction dimension
K: for reduction dimension

Then this class need to be renamed to
GridwiseReduction_mk_to_m_blockwise

Reply:
Yes,   I have considered to change `xy_to_x` to  `mk_to_m`